data_structures/linked_list/cycle_in_linked_list.c

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file
+ * @author [Ankita Kulkarni](https://github.com/ankitakul)
+ * @brief This Algorithm detects a loop in a singly linked list 
+*/
+
+#include <stdio.h>  /// for IO operations (`printf`)
+#include <stdlib.h> /// for the `malloc` operation
+ 
+/**
+* @brief Linked list node structure
+*/
+struct node {
+  int data; //data assigned to the data part of the node
+  struct node *next; //pointer to next node in linked list
+} *head;
+ 
+ 
+/**
+ * @brief Creates a new node and inserts it in front of a singly linked list
+ * @param num data that is assigned to the node
+ * @returns void
+*/
+
+void insert(int num) {
+    // create a new node
+    struct node* newNode = (struct node*) malloc(sizeof(struct node));
+    
+    // assign value(num)
+    newNode->data  = num;
+    
+    // next pointer of new node will point to head node of linked list
+    newNode->next = head;
+    
+    //head of the list is the new node
+    head = newNode;
+    
+    //printf("Inserted Element : %d\n", num);
+}
+ 
+/**
+ * @brief Checks if a loop exists in the linked list
+ * @param head pointer to the first node in the linked list
+ * @returns void
+*/
+void findLoop(struct node *head) {
+
+    // creating slow and fast pointers.
+    struct node *slow, *fast;
+    slow = fast = head;
+   
+    while(slow && fast && fast->next) {
+        // slow pointer will move one node per iteration  
+        // fast pointer will move two nodes per iteration
+        // loop exist if slow and fast pointers are pointing to same node
+        slow = slow->next;
+        fast  = fast->next->next;
+        
+        // check if the list contains a loop
+        if (slow == fast) {
+           printf("Linked List contains a loop\n");
+           return;
+        }
+    }
+    
+    printf("No Loop in Linked List\n");
+}
+
```

Comment:
Missing self-test implementations (see our [contributing guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#test) for more information).